src/Transport/Topology/Topologies/ForwardingTopologySectionManager.cs

```diff
@@ -89,30 +98,32 @@ public TopologySectionInternal DetermineResourcesToCreate(QueueBindings queueBin
             };
         }
 
-        public TopologySectionInternal DeterminePublishDestination(Type eventType)
+        public async Task<TopologySectionInternal> DeterminePublishDestination(Type eventType)
```

Comment:
On the fence about this. Maybe we need to brainstorm here about the consequences of this change. GetOrAdd could be executed multiple times but only one result would be added. The call offerred some "atomicity" but this new version splits it into multiple calls. It is possible to combine Lazy or AsyncLazy with the dictionary if we do it right. The question is whether it is necessary.

@Scooletz thoughts?